backend/schedulerd/proxy_check.go

```diff
@@ -18,13 +19,22 @@ func matchEntities(entities []*types.Entity, proxyRequest *types.ProxyRequests)
 
 OUTER:
 	for _, entity := range entities {
-		for _, expression := range proxyRequest.EntityAttributes {
+		for _, statement := range proxyRequest.EntityAttributes {
 			parameters := map[string]interface{}{"entity": entity}
 
-			result, err := eval.EvaluatePredicate(expression, parameters)
+			result, err := eval.EvaluatePredicate(statement, parameters)
 			if err != nil {
+				fields := logrus.Fields{"statement": statement}
+				// Only report an error if the expression's syntax is invalid
+				switch err.(type) {
+				case eval.SyntaxError:
+					logger.WithFields(fields).WithError(err).Error("syntax error")
+				case eval.TypeError:
+					logger.WithFields(fields).WithError(err).Error("type error")
+				default:
+					logger.WithFields(fields).WithError(err).Debug("missing attributes")
```

Comment:
Maybe we could take this opportunity to add the entity name (with its org & env) in the fields of these log statements? Otherwise I realize it's a bit hard to figure out which entities are missing attributes:

```
{"component":"schedulerd","error":"failed to evaluate the expression: input is empty, get [Team] from invalid","level":"debug","msg":"missing attributes","statement":"entity.Team == 'ops'","time":"2018-06-19T10:32:40-04:00"}
{"component":"schedulerd","error":"failed to evaluate the expression: input is empty, get [Team] from invalid","level":"debug","msg":"missing attributes","statement":"entity.Team == 'ops'","time":"2018-06-19T10:32:40-04:00"}
{"component":"schedulerd","error":"failed to evaluate the expression: input is empty, get [Team] from invalid","level":"debug","msg":"missing attributes","statement":"entity.Team == 'ops'","time":"2018-06-19T10:32:40-04:00"}
{"component":"schedulerd","error":"failed to evaluate the expression: input is empty, get [Team] from invalid","level":"debug","msg":"missing attributes","statement":"entity.Team == 'ops'","time":"2018-06-19T10:32:40-04:00"}
{"component":"schedulerd","error":"failed to evaluate the expression: input is empty, get [Team] from invalid","level":"debug","msg":"missing attributes","statement":"entity.Team == 'ops'","time":"2018-06-19T10:32:40-04:00"}
```

Also, in addition to telling the user that the entity is missing attributes, I suggest we also specify what we are going to do about it (i.e. ignore the entity) if it makes sense.

Reply:
Great suggestion!